appinventor/appengine/src/com/google/appinventor/server/storage/ObjectifyStorageIo.java

```diff
@@ -944,6 +946,39 @@ public void run(Objectify datastore) {
   }
 
   @Override
+  public List<UserProject> getUserProjects(final String userId, final List<Long> projectIds) {
+    final Result<Map<Long,ProjectData>> projectDatas = new Result<Map<Long,ProjectData>>();
+    try {
+      runJobWithRetries(new JobRetryHelper() {
+        @Override
+        public void run(Objectify datastore) {
+          Map<Long,ProjectData> pd = datastore.get(ProjectData.class, projectIds);
+          if (pd != null) {
+            projectDatas.t = pd;
+          } else {
+            projectDatas.t = null;
+          }
+        }
+      }, false);
+    } catch (ObjectifyException e) {
+      throw CrashReport.createAndLogError(LOG, null,
+          collectUserErrorInfo(userId), e);
+    }
+    if (projectDatas.t == null) {
+      return null;
```

Comment:
+1 as well. This changes the contract behavior relative to the lines removed from [ProjectServiceImpl.java](https://github.com/mit-cml/appinventor-sources/pull/836/files#diff-dacdac42ba5eb0de1b23d5ec187829daL243) that would return an empty list on failure.